provisioning/transport/http/src/http.ts

```diff
@@ -0,0 +1,271 @@
+      if (err) {
+        callback(err);
+      } else {
+        callback(null, responseBody);
+      }
+    });
+
+    /* TODO: Codes_SRS_NODE_PROVISIONING_HTTP_18_031: [ If `disconnect` is called while the registration request is in progress, `register` shall call the `callback` with an `OperationCancelledError` error. ] */
+  }
+
+  private _sendOperationStatusRequest(callback: (err: Error, body: any) => void): void {
+    /* Codes_SRS_NODE_PROVISIONING_HTTP_18_022: [ operation status request polling shall be a GET operation sent to 'https://global.azure-devices-provisioning.net/{idScope}/registrations/{registrationId}/operations/{operationId}' ] */
+    /* Codes_SRS_NODE_PROVISIONING_HTTP_18_037: [ The operation status request shall include the current `api-version` as a URL query string value named 'api-version'. ] */
+    let path: string = '/' + this._idScope + '/registrations/' + this._registrationId + '?api-version=' + this._apiVersion;
+
+    /* Codes_SRS_NODE_PROVISIONING_HTTP_18_020: [ The operation status request shall have the following in the Http header:
+      Accept: application/json
+      Content-Type: application/json; charset=utf-8 ] */
+    let httpHeaders = {
+      'Accept' : 'application/json',
+      'Content-Type' : 'application/json; charset=utf-8'
+    };
+
+    /* Codes_SRS_NODE_PROVISIONING_HTTP_18_021: [ If an `authorization` string is specifed, it shall be URL encoded and included in the Http Authorization header of the operation status request. ] */
+    /* Codes_SRS_NODE_PROVISIONING_HTTP_18_023: [ If the operation status request times out, `register` shall stop polling and call the `callback` with a `TimeoutError` error. ] */
+    /* Codes_SRS_NODE_PROVISIONING_HTTP_18_024: [ `register` shall deserialize the body of the operation status response into an object. ] */
+    /* Codes_SRS_NODE_PROVISIONING_HTTP_18_025: [ If the body of the operation status response fails to deserialize, `register` will throw an `SyntaxError` error. . ] */
+    /* Codes_SRS_NODE_PROVISIONING_HTTP_18_026: [ If the operation status response contains a failure status code, `register` shall stop polling and call the `callback` with an error created using `translateError`. ] */
+    this._currentHttpRequest = this._restApiClient.executeApiCall('GET', path, httpHeaders, this._body, this._httpTimeout, (err: Error, responseBody?: any, response?: any) => {
+      if (err) {
+        callback(err, null);
+      } else {
+        callback(null, responseBody);
+      }
+    });
+
+    /* TODO: Codes_SRS_NODE_PROVISIONING_HTTP_18_032: [ If `disconnect` is called while the operation status request is in progress, `register` shall call the `callback` with an `OperationCancelledError` error. ] */
+  }
+
+  private _safeCallback(callback: (error2?: Error | null, result2?: any) => void, error?: Error | null, result?: any): void {
```

Comment:
>_safeCallback [](start = 10, length = 13)

do we need this? all callbacks should be mandatory. I would like to push the "optional callback" only on the highest layer of the user-accessible API

Reply:
agreed

---
In reply to: [141479907](https://github.com/Azure/azure-iot-sdk-node/pull/93#discussion_r141479907) [](ancestors = 141479907)